src/github/pullRequestManager.ts

```diff
@@ -582,15 +582,81 @@ export class PullRequestManager {
+			const { remote, octokit } = await githubRepository.ensure();
+			const currentUser = octokit && (octokit as any).currentUser;
+			const currentUserLogin: string = currentUser.login;
+
+			if(!!pullRequestResponseData) {
+				pageInformation.hasMorePages = pullRequestResponseData.repository.pullRequests.pageInfo.hasNextPage;
+				pageInformation.endCursor = pullRequestResponseData.repository.pullRequests.pageInfo.endCursor;
+			} else {
+				pageInformation.hasMorePages = false;
+				pageInformation.endCursor = null;
+			}
+
+			if (pullRequestResponseData && pullRequestResponseData.repository.pullRequests.nodes.length) {
+				let pullRequestItems = pullRequestResponseData.repository.pullRequests.nodes;
 
-			pageInformation.hasMorePages = !!pullRequestData && pullRequestData.hasMorePages;
-			pageInformation.pullRequestPage++;
+				if (type !== PRType.All) {
+
+					let pullRequestFilter: (item: PullRequestListItem) => boolean;
+
+					if (type === PRType.Mine) {
+						pullRequestFilter = pr => pr.author.login === currentUserLogin;
+					} else if (type === PRType.RequestReview) {
+						pullRequestFilter = pr => pr.reviewRequests.nodes
+							.findIndex(reviewRequest => reviewRequest.requestedReviewer.login === currentUserLogin) !== -1;
+					} else if (type === PRType.AssignedToMe) {
+						pullRequestFilter = pr => pr.assignees.nodes
+							.findIndex(asignee => asignee.login === currentUserLogin) !== -1;
+					} else {
+						throw new Error('Unexpected pull request filter');
+					}
+
+					pullRequestItems = pullRequestItems.filter(pullRequestFilter);
+				}
+
+				const pullRequests: PullRequestModel[] = pullRequestItems.map(pullRequestItem => {
+						let assignee: IAccount | undefined;
+						if(!!pullRequestItem.assignees.nodes && pullRequestItem.assignees.nodes.length) 						{
+							assignee = pullRequestItem.assignees.nodes[0];
+						}
+
+						const pullRequest: PullRequest = {
+							url: pullRequestItem.url,
+							assignee,
+							base: {
+								label: `${pullRequestItem.baseRef.repo.owner}:${pullRequestItem.baseRef.name}`,
+								ref: pullRequestItem.baseRef.name,
+								repo: {cloneUrl: pullRequestItem.baseRef.repo.url },
+								sha: pullRequestItem.baseRef.target.sha
+							},
+							head: {
+								label: `${pullRequestItem.headRef.repo.owner}:${pullRequestItem.headRef.name}`,
+								ref: pullRequestItem.headRef.name,
+								repo: {cloneUrl: pullRequestItem.headRef.repo.url },
+								sha: pullRequestItem.baseRef.target.sha
+							},
+							labels: [],
+							body: '',
+							createdAt: pullRequestItem.createdAt,
+							updatedAt: pullRequestItem.updatedAt,
+							merged: pullRequestItem.merged,
+							nodeId: pullRequestItem.nodeId,
+							number: pullRequestItem.number,
+							state: pullRequestItem.state,
+							title: pullRequestItem.title,
+							user: pullRequestItem.author
+						};

```

Comment:
Mapping the graphql object to an object the `PullRequestModel` consumes. Not thrilled about doing this here.

Reply:
the file `src/github/utils.ts` has functions for mapping response types to normalized types

Reply:
Thanks, i'll move the code.